keps/77-dynamically-sized-jobs/README.md

```diff
@@ -0,0 +1,286 @@
+  - [User Stories](#user-stories)
+    - [Story 1 - RayCluster w/ autoscaling](#story-1---raycluster-w-autoscaling)
+- [Design Details](#design-details)
+  - [Workload Slices](#workload-slices)
+  - [Creating Workload Slices](#creating-workload-slices)
+  - [Pod Scheduling Gates](#pod-scheduling-gates)
+  - [Garbage Collecting Workload Slices](#garbage-collecting-workload-slices)
+- [Phases for MVP (alpha)](#phases-for-mvp-alpha)
+  - [Phase 1 - Scale Down](#phase-1---scale-down)
+    - [Job controller](#job-controller)
+  - [Phase 2 - Aggregating Workload Slices](#phase-2---aggregating-workload-slices)
+  - [Phase 3 - Scale up with Workload Slices and Scheduling Gates](#phase-3---scale-up-with-workload-slices-and-scheduling-gates)
+    - [Scheduler](#scheduler)
+- [Additional Details](#additional-details)
+  - [Feature Gate](#feature-gate)
+  - [Locking Flavor Assignments for Workload Slices](#locking-flavor-assignments-for-workload-slices)
+  - [Webhook changes](#webhook-changes)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Ignore Resize from Kuberay](#ignore-resize-from-kuberay)
+<!-- /toc -->
+
+## Summary
+
+Enable dynamic sizing of Kueue jobs. For the MVP, we will only focus on supporting RayClusters with autoscaling enabled, but other resources that can benefit from dynamic sizing should be supported eventually.
+
+See: [Support dynamically sized (elastic) jobs #77](https://github.com/kubernetes-sigs/kueue/issues/77)
+
+## Motivation
+
+Kueue currently lacks support for resizing jobs. When a job is resized, Kueue will recreate the Workload representation of the job, leading to a disruptive suspend and requeue process. This limitation hinders the usability of Kueue for resources like RayCluster that sometimes have autoscaling capabilities enabled.
+
+To properly support RayCluster, Kueue needs to gracefully handle the scale up and scale down of RayCluster nodes. Concretely this means that scaling the resources used by a RayCluster is appropriately reflected in the respective ClusterQueue without needing to suspend the entire cluster.
+
+### Goals
+
+- Gracefully handle resize operations for Kueue jobs (i.e. update quota usage without suspend, enqueue scale ups)
+- Autoscaling RayCluster works with Kueue (MVP)
+
+### Non-Goals
+
+- Vertical scaling of workloads – Kueue will only handle resize operations that scale Pods horizontally 
+- Support resize for other Kueue jobs such as QueuedWorkload, JobSet, etc (future)
+- Resizing of RayJobs
+- Partial Preemption
+
+## Proposal
+
+Update the Job framework reconciler and introduce new controllers to orchestrate dynamic resizing of jobs. We are only interested in horizontal scaling of jobs (e.g. scaling more replicas). At a high level, this will be accomplished by:
```

Comment:
Why new controllers are needed (instead of just updating the existing ones)?

Reply:
To have a clear separation of tasks that are specific to resizing a job: create workload slice(s) -> aggregate workload slice(s) -> ungate pods